### ISSUE TITLE
Zero-based counting in "train epoch" progress bar.

### ISSUE THREAD

**Issue by PhilipMay** [deepset-ai/FARM#381]
The train progress bar seems to work zero based. See below:
 
```
Train epoch 0/20 (Cur. train loss: 0.5595):  13%|█▎        | 368/2813 [10:44<48:51,  1.20s/it]
```

IMO this is counterintuitive. It should count from 1 to 20 in this case (or from 0 to 19) since I said `n_epochs=20` in `initialize_optimizer`.

What do you think?

**Comment by Timoeller** [comment]
Totally agree. Please feel free to add a PR.

**Comment by tholor** [comment]
I remember that we had quite some discussion about this a while ago. The zero-based counting simplified some functionality around Checkpointing, StreamingDataSilo and LR schedules. 
I would vote for keeping it like it is at least until we merge #305 as there might be many unwanted side effects otherwise.

**Comment by PhilipMay** [comment]
Well. IMO this is no big deal. I think we should plit 2 questions:
1. How do we count epochs internaly?
2. How do we show them to the user?

I am not 100% sure and I did not try it but I think this could be a simple solution:

Instead of:
` progress_bar.set_description(f"Train epoch {epoch}/{self.epochs} (Cur. train loss: {loss:.4f})")`
we could just say: ` progress_bar.set_description(f"Train epoch {epoch+1}/{self.epochs} (Cur. train loss: {loss:.4f})")` see here: https://github.com/deepset-ai/FARM/blob/master/farm/train.py#L249

IMO this is the same as Keras is also doing. They show the epochs one based but internaly they count them zero based.

What do you think?

**Comment by Timoeller** [comment]
I think this is a good solution and I do not see any problems in other parts of the code when changing this line.

**Comment by PhilipMay** [comment]
@tholor What do you think about my suggestion? Does it interfere with #305 or is it independent?

**Comment by tholor** [comment]
Hm... while I  see your point about a "simple presentation logic", I still have some doubts to be honest: 
- If the epoch count is one-based the step count (within an epoch) should also be switched to one-based and there it might be more than a "logged string" that we need to change
- If we have a different counting logic for "presentation" and "internal", it might become ambiguous for some other parts of the pipeline. For example, how would we name saved checkpoint files - using "internal" or "presentation" logic? As this was a source of bugs in the past, I would be very cautious here (Disclaimer: I am probably biased because I once spent > 1 day tracing a bug where it turned out in the end that we were executing a training step twice after stopping & resuming training using checkpointing).

I would propose to postpone this until #305 is finished & merged and evaluate then if we want to "completely" move to one-based counts.

**Comment by PhilipMay** [comment]
Well - yes. There is pro and con for zero and one based counting. At the end of the day I do not care and can can live with both solutions. If you want to be zero based we should count from 0 to 19 if saying `n_epochs=20`. 

Then we could say `progress_bar.set_description(f"Train epoch {epoch}/{self.epochs-1} (Cur. train loss: {loss:.4f})"`. The current implementation is just counterintuitive IMO.

Maybe you can discuss this internally and make a decision on how to count. Personaly I would definitly count zero based internaly. External I do not care.

**Comment by tholor** [comment]
We had a discussion in the team and everybody was in favor of zero-based counting (interal+external). 
We like your suggestion though to switch the progress bar to `progress_bar.set_description(f"Train epoch {epoch}/{self.epochs-1} (Cur. train loss: {loss:.4f})"`. 
Hope this makes sense to you. We appreciate your initiative here and the open discussion.  

Do you want to create a quick PR for that?

**Comment by PhilipMay** [comment]
Yes - I can create a PR for this.
Hope the CI runs soon. :-)

**Comment by PhilipMay** [comment]
PR added: #398
